### PR TITLE
Added 8021q kernel module for VLAN support

### LINKED ISSUE #211: 802.1q VLAN Support
Hi,

I recently hooked up HassOS with KVM on my hypervisor. Since I've different VLAN's that Hass should work with, I set up two new network manager profiles for these, using the single virtio link as parent.

However, it seems that the 8021q kernel module is missing. Would it be possible to include that in future builds? For what it's worth, that's something really useful.

Personally, I can just add another virtio link to the VM and couple network manager profiles to their MACs, but devices with only a single physical NIC don't have that option.

Best regards
Robert

### PR DESCRIPTION
**Pull request by efficiosoft**
Closes #211

### CODE REVIEW

File: buildroot-external/kernel/hassos-4_14.config [hunk 1]
```diff
@@ -41,6 +41,10 @@ CONFIG_MACVLAN=y
 CONFIG_IPVLAN=y
 CONFIG_VXLAN=y
 
+CONFIG_VLAN_8021Q=m
```
Comment (pvizeli) [inline buildroot-external/kernel/hassos-4_14.config]: Will that autoload my NetworkManager?

**Comment by efficiosoft** [comment]
If some VLAN should be used or not has to be configured in the networkmanager profiles, as if you want to configure a WiFi connection. But yes, ``ip link add link eth0 type vlan id 123`` is enough to have the ``8021q`` module loaded automatically.

**Comment by scottini** [comment]
Hello, how can I update my Hass OS installation to enable VLAN support? Then, how can I configure the functionality and, above all, make the config permanent?

**Comment by efficiosoft** [comment]
@scottini Either build it yourself with the changes I made (not recommended if you are completely unfamiliar with those sort of things) or wait until it gets released.

Anyway, you'll find how to create VLAN connections in the NetworkManager docs. After all, it's just plain NM connection profiles. The installation tutorial mentions how to provide external connection profiles to hassio.

**Pull request merged**